broker/src/main/java/org/apache/rocketmq/broker/processor/AbstractSendMessageProcessor.java

```diff
@@ -109,7 +109,7 @@ protected MessageExtBrokerInner buildInnerMsg(final ChannelHandlerContext ctx,
         final SendMessageRequestHeader requestHeader, final byte[] body, TopicConfig topicConfig) {
         int queueIdInt = requestHeader.getQueueId();
         if (queueIdInt < 0) {
-            queueIdInt = Math.abs(this.random.nextInt() % 99999999) % topicConfig.getWriteQueueNums();
```

Comment:
Why was Math.abs deleted？

Reply:
It's meaningless there, I have related it to an Issue

Reply:
ThreadLocalRandom.current().nextInt() and random.nextInt() may produce negative numbers, if you remove Math.abs, there will be problems.

Reply:
As the java doc of Random, it show be positive always
![image](https://user-images.githubusercontent.com/16242100/128480577-fac14815-1bcf-4847-966e-adabb41935c5.png)